File: KNURE-Sked/Lesson.m
//
//  Lesson.m
//  KNURE-Sked
//
//  Created by Vlad Chapaev on 24.12.15.
//  Copyright © 2015 Shogunate. All rights reserved.
//

#import "Lesson.h"

@implementation Lesson

- (id)initWith:(NSString *)title
          name:(NSString *)fullName
      auditory:(NSString *)auditory
          type:(LessonType)type
     startDate:(NSDate *)startDate
       endDate:(NSDate *)endDate
   teacherName:(NSString *)teacherName
        groups:(NSArray *)groups {
    
    self = [super init];
    if (self) {
        self.title = title;
        self.fullName = fullName;
        self.auditory = auditory;
        self.type = type;
        self.startTime = startDate;
        self.endTime = endDate;
        self.teacherName = teacherName;
        self.groups = groups;
    }
    return self;
}

@end

File: KNURE TimeTable iOS/DomainLayer/entities/Lesson/Lesson.swift
//
//  Lesson.swift
//  KNURE TimeTable iOS
//
//  Created by Vladislav Chapaev on 23/02/2019.
//  Copyright © 2019 Vladislav Chapaev. All rights reserved.
//

import Foundation

struct Lesson: Sendable {

//	let auditory: String
//
//	let start: Date
//
//	let end: Date
//
//	let number: Int
//
//	let subject: Subject
}

extension Lesson {

	struct Subject: Sendable {

		let identifier: String

		let brief: String

		let title: String
	}
}
